### DISCUSSION TITLE
Meeting: Celo Wallet and (d)App Council #10 [April] 💬

### DISCUSSION THREAD

**Discussion by arthurgousset** [celo-org/celo-monorepo#9426] [Wallet and (d)App Council]
### Join

- [Google Meet](https://meet.google.com/uwj-wfse-uzm)
- [Recording](https://drive.google.com/file/d/1Pf93D3NGqM0uTENiqiZVr-asYxMpOJ48/view?usp=sharing) 

### Useful links

- Notion page with meeting notes: [Celo Wallet Council Meeting Notes](https://www.notion.so/celoplatform/Celo-Wallet-Council-61dfe28e98a543409cf6c181b132c4de)
- Last meeting: https://github.com/celo-org/celo-monorepo/discussions/9365
- Forum post: https://forum.celo.org/t/celo-wallet-and-d-app-council-10-april/3413
--------------------------------------------

Hey all :wave:

The monthly Celo Wallet and (d)App Council meeting is taking place on **Wednesday, April 27th (9am PDT/6pm CEST)**.

## 🤷‍♂️ What is this?

A monthly meeting for Celo **developers, founders,** **product** people (anyone really!) to come together and
- find consensus (coordinate),
- learn about needs, pain points, and desires
- present and collect technical feedback

## 🗒️  Agenda

- Comment with a short sentence describing what you'd like to discuss
- Don't worry, it can be small or big! 
- Every builder is welcome (and we are very easy going, I promise 😄)
- Upvote topics you're interested in

## Housekeeping:

- [x] Send update calendar invite with Github link
- [x] Add agenda items to vote on
- [x] Publicise on forum and Discord

--------------------------------------------

### Proposed agenda items 👇  

Please feel free to add agenda items (however small or big!) below.
 
For example:
- broad discussion topics to **amplify** your ideas/features/frictions, 
- requests for **feedback** (from the community), 
- roadmap **updates**,
- etc (anything that would be useful for you really!) 

🗳️ You can upvote/downvote agenda items below.

---

### Reminders

- [x]  Appoint secretary ✍️
- [x]  RECORD meeting 🎥

**Comment by nembal** [comment]
Agenda item proposal: 

- 👀 kycDAO status report and next steps

**Reply by arthurgousset** [reply]
### Meeting notes

- KycDAO takes care of compliance and regulatory reporting
- uses non-transferable NFTs (721 without transfer function)
- coming to Celo relatively soon
- resource: https://blog.kycdao.xyz/nft_vs_vc/
- service live on Near ecosystem (got a grant)
  - easy contract upgrading
  - but EVM contract, so easy to move
- @nembal: Would anyone be interested in being a prototype user? 
  - @cajubelt would like to try
  - Jesse keen to try out too

Sign up using Google form: https://forms.gle/k5JmzrTF9a2aJoDJ7

**Reply by arthurgousset** [reply]
- Aaron: Is Near/Celo communication easy?
 - Celo <-> Polygon is relatively easy (similar addresses)

**Comment by arthurgousset** [comment]
Temperature check on **Plumo** (light-client): 
- which wallets are interested in integrating plumo 
- if not why not

**Reply by arthurgousset** [reply]
## Meeting notes

@mstraka100: 
- technical update
  - have circuit, setup and prover code done
  - still work outstanding on proof gossiping protocol
- wallet integration
  - basically only requires proof verification (to verify proofs the prover creates)

@silasbw:
- Valora hasn't thought about Plumo all that much. 
- Would like to understand the impact on users better

Ronan (Trelis): 
- it's light to run but heavy to download
- @mstraka100: 
  - Plumo is good at light verification, but the tradeoff is data consumption for downloading the Plumo client
  - We might need to do more work to reduce the size of the Plumo client
- we're thinking of this from the angle of running a node (but will probably run a full node)

@jeanregisser: 
- would be great to hear from @jmrossy (maybe in the future) because he worked on the web integration for plumo.

@aaronmgdr: 
- Celo web wallet has currenlty integrate Plumo in the browser
_Edit: to account for nuance @aaronmgdr provided below._

Ronan: 
- Would be great to have a written blurb about the pros/cons of Plumo and some of the hard numbers (cc @mstraka100)

**Reply by aaronmgdr** [reply]
hmm the note above sounds like its hard and others tried but failed, however my comment intent that at least one wallet had and im sure more will.

**Reply by aaronmgdr** [reply]
BTW I checked in with the crypto engineer Kobi who led plumo and he confirmed that https://www.npmjs.com/package/plumo-verifier is wasm plumo client 

Its 600k which isn't small but  isn't dramatically big for a Dapp. You could replace contract kit 1.5 with contractkit2 + plumo and still come out ahead. or web3 with ethersjs + plumo and bundle size would not change much

**Reply by arthurgousset** [reply]
@aaronmgdr thanks for clarifying! Unfortunately, I'm sure many nuances get lost in translation when I take brief notes while people talk live (sorry!) 🤷‍♂️ I appreciate the additional context, I reflected that in the notes above 👍

**Comment by aaronmgdr** [comment]
ContractKit 2.0 and associated packages have been released

[release notes](https://github.com/celo-org/celo-monorepo/releases/tag/contractkit-v2.0)

[upgrade guide](https://docs.celo.org/developer-guide/contractkit/migrating-to-contractkit-v2)

**Reply by arthurgousset** [reply]
### Meeting notes

- CK 2.0 has been release (woot woot!)
- Multiple libraries are bumped at the same time (celo/utils, ...)
- did not upgrade use-contractkit yet
- it's fast
- also created mini-contractkit (same interface but doesn't bring in all the contracts)

@jmrossy: big clap for Aaron!!! 
- quick question: how far along are we to include changes in use-contractkit
- @aaronmgdr: there is a branch and it's not a lot of work to upgrade it
  - most dapps don't need the attestations and governance contract
  - tradeoff between releasing full contractkit changes or mini-contractkit changes
  - bigger breaking change so potentially dangerous
- @jmrossy: "Make common things easy, hard things possible"

@aaronmgdr: Biggest changes is: you only need an instance of the celo-connection class. (you only need one wrapper now)

@jeanregisser: valora is already 100MB in size, so size changes are marginal.

**Comment by arthurgousset** [comment]
Primus money sharing mini intro on what they are working on: 
- https://youtu.be/oi2VS9nbaVs
- https://forum.celo.org/t/woocommerce-plugin-for-payments-in-cusd-or-ceur/1286

**Reply by arthurgousset** [reply]
## Meeting notes 

- working on desktop (pay with mobile app)
- mobile to mobile (is difficult today)
- already in touch with Subha (re Payment SDK)

Ronan: What other chains do you support
- Primus Money: Fuse (particularly interested in Celo because of retail payments)

**Comment by arthurgousset** [comment]
Jesse intro: 
- working as a partner engineer at cLabs
- [contact him](mailto:jesse.sawa@clabs.co) if you need help!

**Comment by arthurgousset** [comment]
[~40min onwards]

**Ronan: [Proof of deposit comment on forum.celo.org](https://forum.celo.org/t/proof-of-deposit-initial-proposal-and-request-for-feedback/3041/122) and community wide rewards (supercharge)**
- it's a bit like UBI (**distributing rewards per person instead of per dollar**)
- interesting link to Valora supercharge rewards
- "wouldn't it be nice if we had ecosystem wide incentive rewards"

**Reply by arthurgousset** [reply]
## Meeting notes

Denisse: 
- I've been thinking a lot about Celo rewards
- Original celo rewards was wallet agnostic (CGP 14)
- But founding run (ran for 1.5 years)
- Currently supercharge reward is funded by Valora (exclusively)
- Because our funding is constrained (short-term), extremely interested in connecting with other wallets that are interested in participating in celo rewards proposal
- Link: https://forum.celo.org/t/governance-proposal-to-reward-early-users/662/

OpenCelo: 
- first initiative was separate from CCF1, but Valora was the only one that got the benefit from this, because it's the only wallet that used phone number verification at the time

Ronan: 
- phone number verification seems useful
- OpenCelo: 
  - ImpactMarket doesn't require phone number verification for UBI atm
  - I can't commit today that I'd like to see rewards again
  - I want to actually see user counts go up
  - Otherwise it's the same thing as DeFi For The People

Denisse: 
- Phone numbers are very useful for sybil resistance (stopping whales and bots)
- We are working on providing an update on user adoption
- When the rate changed from 50% to 25%, we didn't see supercharge users drop

OpenCelo: 
- I would like to see more regular updates on celo rewards (more than 6mo)
- Something I'd rather see is a **cash-back system** (e.g. select merchants: Fireblocks)
- [ ] Possibly link past updates

Denisse: 
- Initially CGP was focused on stablecoin adoption (globally)
- Focusing on payments is basically focusing on the global north
- OpenCelo: 
  - FirstDAG is entirely focused on emerging markets

Ronan: 

> Just a forum post I've written here that I think is important to wallets with respect to cStable collateralisation/safety: https://forum.celo.org/t/upcoming-cgp-increase-target-level-of-stable-holdings-in-the-reserve/3380/17
> Arthur, I agree that we should have a discussion to find a path forward involving many wallets to talk rewards
